pkg/kubelet/logs/container_log_manager.go

```diff
@@ -213,66 +223,96 @@ func (c *containerLogManager) Clean(ctx context.Context, containerID string) err
-			// lose original container log after ReopenContainerLog fails.
-			// We try to recover it by reopening container log.
-			if err := c.runtimeService.ReopenContainerLog(ctx, id); err != nil {
-				klog.ErrorS(err, "Container log doesn't exist, reopen container log failed", "containerID", id, "path", path)
-				continue
-			}
-			// The container log should be recovered.
-			info, err = c.osInterface.Stat(path)
-			if err != nil {
-				klog.ErrorS(err, "Failed to stat container log after reopen", "path", path)
-				continue
-			}
+		c.queue.Add(container.GetId())
+	}
+	wait.PollInfinite(20*time.Millisecond, func() (done bool, err error) {
+		return c.queue.Len() == 0, nil
+	})
+	klog.V(4).InfoS("Finished container log rotation sequence")
+	return nil
+}
+
+func (c *containerLogManager) processContainer(ctx context.Context, worker int) (ok bool) {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer func() {
+		c.queue.Done(key)
+		c.queue.Forget(key)
+	}()
+	// Always default the return to true to keep the processing of Queue ongoing
+	ok = true
+	id := key.(string)
+
+	resp, err := c.runtimeService.ContainerStatus(ctx, id, false)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get container status", "worker", worker, "containerID", id)
+		return
+	}
+	if resp.GetStatus() == nil {
+		klog.ErrorS(err, "Container status is nil", "worker", worker, "containerID", id)
+		return
+	}
+	path := resp.GetStatus().GetLogPath()
+	info, err := c.osInterface.Stat(path)
+
+	if err != nil {
+		if !os.IsNotExist(err) {
+			klog.ErrorS(err, "Failed to stat container log", "worker", worker, "containerID", id, "path", path)
+			return
 		}
-		if info.Size() < c.policy.MaxSize {
-			continue
+
+		if err = c.runtimeService.ReopenContainerLog(ctx, id); err != nil {
+			klog.ErrorS(err, "Container log doesn't exist, reopen container log failed", "worker", worker, "containerID", id, "path", path)
+			return
 		}
-		// Perform log rotation.
-		if err := c.rotateLog(ctx, id, path); err != nil {
-			klog.ErrorS(err, "Failed to rotate log for container", "path", path, "containerID", id)
-			continue
+
+		info, err = c.osInterface.Stat(path)
+		if err != nil {
+			klog.ErrorS(err, "Failed to stat container log after reopen", "worker", worker, "containerID", id, "path", path)
+			return
 		}
 	}
-	return nil
+	if info.Size() < c.policy.MaxSize {
+		klog.V(7).InfoS("log file doesn't need to be rotated", "worker", worker, "containerID", id, "path", path, "currentSize", info.Size(), "maxSize", c.policy.MaxSize)
+		return
+	}
+
+	if err := c.rotateLog(ctx, worker, id, path); err != nil {
+		klog.ErrorS(err, "Failed to rotate log for container", "worker", worker, "containerID", id, "path", path, "currentSize", info.Size(), "maxSize", c.policy.MaxSize)
+		return
+	}
+	return
 }
 
-func (c *containerLogManager) rotateLog(ctx context.Context, id, log string) error {
+func (c *containerLogManager) rotateLog(ctx context.Context, worker int, id, log string) error {
```

Comment:
Looks like `worker int` isn't used anymore?

Reply:
Ack. Cleaning it up.